### DISCUSSION TITLE
Material assignment (override) in USD hierarchy

### DISCUSSION THREAD

**Discussion by yolov** [Autodesk/maya-usd#1421] [General]
I just wanted to ask if this is a correct workflow:
Create a USD stage from file > expand hierarchy > assign a maya material onto an object from the hierarchy.

I wonder, as comparing a maya scene before/after such an assignment yields difference for defnind the material in the maya scene, but the scene holds no information on how this material is assigned (although opening the scene renders as expected).
Also - selecting anything down in the hierarchy, returns an empty selection, when the selection is queried.
And finally - I've had to kill maya a few times, as it would just freeze if I do that with a standardSurface material.

Thanks in advance to anyone with any information on this.

(
repro steps:
1. create a sphere
2. export to usd
3. create usd stage with the sphere
4. outliner > enable shapes
5. expand usd hierarchy
6. select the second object pSphereShape in usd hierarchy
7. assign standardSurface
8. deselect sphere and re-select sphere

Maya 2022 with MayaUSD 0.9.0
)

**Comment by yolov** [comment]
Just an update that might make things a bit more clear - I can assign a material onto the usdProxy node and it will act as a 'global override' for all objects inside the USD file. And I'm trying to figure out if
1. this is an intended workflow - is the user supposed to do this?
2. is there any way provided (or planned for) for the user to have a way to assign materials from the maya scene onto the different objects of the USD file through the mayaUsd proxy node?

Thanks in advance,
Alex

**Comment by wtelford** [comment]
Hey Alex,
Maya USD does not yet have a material authoring workflow. In addition the workflow you are describing is surprising to me, as I would not expect that to work. That behavior sounds like a bug and I am unable to repro with the steps you provided. Could you provide more detailed steps or even a video?

While we don't have any current material authoring workflows, it might be possible for you via script to implement a partial material authoring workflow using Maya USD's existing import/export functionality. You can look at @kxl-adsk's  post here on how to explore that possibility.
[https://github.com/Autodesk/maya-usd/discussions/1054#discussioncomment-641407](https://github.com/Autodesk/maya-usd/discussions/1054#discussioncomment-641407)

**Reply by yolov** [reply]
Thanks for confirming this for me, Will.
I'll look at the post, you're linking!

Alex
